lib/run.js

```diff
@@ -56,18 +58,24 @@ function runFirefox (options) {
   return normalizeBinary(options.binary).then(function(binary) {
     // Using `spawn` so we can stream logging as they come in, rather than
     // buffer them up until the end, which can easily hit the max buffer size.
-    var firefox = spawn(binary, args, { env: env });
-
-    firefox.on("close", function () {
-      process.removeListener("exit", killFirefox);
-    });
+    var firefox = spawn(binary, args, { env: env, detached: detached });
 
     function killFirefox () {
       firefox.kill();
     }
 
-    // Kill child process when main process is killed
-    process.once("exit", killFirefox);
+    if (!detached) {
+      firefox.on("close", function () {
+        process.removeListener("exit", killFirefox);
+      });
+
+      // Kill child process when main process is killed
+      process.once("exit", killFirefox);
+    } else {
+      firefox.stdout.destroy();
+      firefox.stdin.destroy();
+      firefox.stderr.destroy();
```

Comment:
Though I still think destroying these streams should be done in caller side, to use this 'detached' option in web-ext, we need https://github.com/facebook/flow/pull/5988 in any way,  but it's not clear to me when the PR for flow gets merged.  So I've changed to destroy the streams here.